Modules/termios.c

```diff
@@ -232,13 +240,19 @@ termios_tcsetattr_impl(PyObject *module, int fd, int when, PyObject *term)
 
     int i;
     PyObject *v;
+    long lng;
     for (i = 0; i < NCCS; i++) {
         v = PyList_GetItem(cc, i);
 
         if (PyBytes_Check(v) && PyBytes_Size(v) == 1)
             mode.c_cc[i] = (cc_t) * PyBytes_AsString(v);
-        else if (PyLong_Check(v))
-            mode.c_cc[i] = (cc_t) PyLong_AsLong(v);
+        else if (PyLong_Check(v)) {
+            lng = PyLong_AsLong(v);
+            if (lng == -1 && PyErr_Occurred()) {
+                return NULL;
+            }
+            mode.c_cc[i] = (cc_t) l;
```

Comment:
I think you need to change this too:

```suggestion
            mode.c_cc[i] = (cc_t) lng;
```

Also, you can declare the variable in this inner scope. It does not have to be outside the for loop.